iot_verifier/src/reward_share.rs

```diff
@@ -329,66 +340,68 @@ impl GatewayShares {
-        // compute the awards per hotspot
+
         tracing::info!(
             %total_dc_shares,
             %total_dc_transfer_rewards_used,
             %dc_transfer_rewards_unused,
             %dc_transfer_rewards_per_share,
             "data transfer rewards"
         );
+        self.beacon_rewards_per_share = beacon_rewards_per_share;
+        self.witness_rewards_per_share = witness_rewards_per_share;
+        self.dc_transfer_rewards_per_share = dc_transfer_rewards_per_share;
+        self.total_rewards_for_poc_and_dc =
+            total_beacon_rewards + total_witness_rewards + total_dc_transfer_rewards_capped;
+    }
+
+    pub fn into_iot_reward_shares(
+        self,
+        reward_period: &'_ Range<DateTime<Utc>>,
+    ) -> impl Iterator<Item = (u64, proto::IotRewardShare)> + '_ {
         self.shares
             .into_iter()
-            .map(move |(hotspot_key, reward_shares)| proto::GatewayReward {
-                hotspot_key: hotspot_key.into(),
-                beacon_amount: compute_rewards(
-                    beacon_rewards_per_share,
-                    reward_shares.beacon_shares,
-                ),
-                witness_amount: compute_rewards(
-                    witness_rewards_per_share,
-                    reward_shares.witness_shares,
-                ),
-                dc_transfer_amount: compute_rewards(
-                    dc_transfer_rewards_per_share,
-                    reward_shares.dc_shares,
-                ),
+            .map(move |(hotspot_key, reward_shares)| {
+                let beacon_amount =
+                    compute_rewards(self.beacon_rewards_per_share, reward_shares.beacon_shares);
+                let witness_amount =
+                    compute_rewards(self.witness_rewards_per_share, reward_shares.witness_shares);
+                let dc_transfer_amount =
+                    compute_rewards(self.dc_transfer_rewards_per_share, reward_shares.dc_shares);
+                proto::GatewayReward {
+                    hotspot_key: hotspot_key.into(),
+                    beacon_amount,
+                    witness_amount,
+                    dc_transfer_amount,
+                }
             })
             .filter(|reward_share| {
                 reward_share.beacon_amount > 0
                     || reward_share.witness_amount > 0
                     || reward_share.dc_transfer_amount > 0
             })
-            .map(|gateway_reward| proto::IotRewardShare {
-                start_period: reward_period.start.encode_timestamp(),
-                end_period: reward_period.end.encode_timestamp(),
-                reward: Some(ProtoReward::GatewayReward(gateway_reward)),
+            .map(|gateway_reward| {
+                let total_gateway_reward = gateway_reward.dc_transfer_amount
+                    + gateway_reward.beacon_amount
+                    + gateway_reward.witness_amount;
+                (
+                    total_gateway_reward,
```

Comment:
does including this total serve any purpose other than testing? wouldn't it be cleaner to just grab the reward message and sum the three reward categories in the test if that's the only place it's used?

Reply:
this is used in the rewarder, to accumulate the final u64 value of each gateway reward and from this to work out the final unallocated reward amount